sync2jira/intermediary.py

```diff
@@ -277,33 +278,37 @@ def map_fixVersion(mapping, issue):
         issue["milestone"] = fixVersion_map.replace("XXX", issue["milestone"])
 
 
-def matcher(content, comments):
+JIRA_REFERENCE = re.compile(r"Relates +to +JIRA:\s*(\w+-\d+)")
+
+
+def matcher(content: Optional[str], comments: list[dict[str, str]]) -> str:
     """
     Helper function to match to a JIRA
 
+    Extract the Jira ticket reference from the first instance of the magic
+    cookie (e.g., "Relates to JIRA: FACTORY-1234") found when searching
+    through the comments in reverse order.  If no reference is found in the
+    comments, then look in the PR description.  This ordering allows later
+    comments to override earlier ones as well as any reference in the
+    description.
+
     :param String content: PR description
     :param List comments: Comments
     :return: JIRA match or None
     :rtype: Bool
     """
-    # Build out a string with all comments and initial_comment
-    all_data = " "
+
+    def find_it(input_str: str) -> str:
+        match = JIRA_REFERENCE.search(input_str)
+        return match.group(1) if match else None
+
     for comment in reversed(comments):
-        all_data += f" {comment['body']}"
-    if content:
-        all_data += content
```

Comment:
@ralphbean, is the `content` (i.e., the PR description) really supposed to be the _last_ resort, such that matches in the comments override it?  It seems like the description (which should be editable after the fact) should rule over the comments...but, I've retained this behavior.  Let me know if I should change it.

Reply:
Yeah, let's retain it for now. It's kind of screwy, but if its what current users expect, let's not upend it without consulting them.